Add seed support for LTI chimes

jxjj opened the pull request:
Currently, the database is seeded with chimes, but these chimes aren't linked to any user (none are seeded), and none of them are LTI flavored. This means testing ChimeIn for LTI linked chimes requires manual setup (and re-setup if you run cypress tests locally when it nukes the database).

So, this PR changes seeding a bit:
- A default admin user is added when the DB is seeded
- The admin user will have 2 seeded chimes attached: one LTI chime and one vanilla chime

Thus:
- A UserSeeder added
- Chime factory now has a `withLTI()` method that can be invoked to turn the chime into an LTI-flavored chime.
- UserFactory and ChimeFactory have been refactored to be in [Laravel 8 factory class](https://laravel.com/docs/8.x/releases#model-factory-classes) style.
- User and Chime classes have the `HasFactory` attribute added.

Meanwhile, two related changes:
- `selectJoinUrl` would throw an error if the `lti_return_url` doesn't contain `/courses/:courseId`, causing the ChimeCard 
to not display a Chime. (Discovered when I was generating random urls with faker). Now, we fall back to the `lti_return_url` if the LMS course url can't be parsed from the `lti_return_url`.
- The `admin/users` table used to show the iteration number rather than the user's id under the `#` column. I changed it to user id, thinking it would be more useful for admins – at least it would be for me. 😄 
- Symfony keeps asking thanks plugin if I run `composer dumpautoload`. I set this to be ignored in composer... not that I'm not thankful...

cmcfadden commented on database/factories/UserFactory.php:
```diff
@@ -13,12 +11,28 @@
+      "name" => $this->faker->name,
+      "email" => $this->faker->unique()->safeEmail,
+      "password" =>
+        '$2y$10$TKh8H1.PfQx37YgCzwiKb.KjNyWgaHb9cbcoQgdIVFlYg7B77UdFm', // secret
```
how'd you know my password?

jxjj replied:
Ha!

cmcfadden commented on resources/assets/js/helpers/chimeSelectors.js:
```diff
@@ -9,7 +9,7 @@ export function selectCanvasCourseUrl(chime) {
   if (!url) return null;
   const found = url.match(/^(?<courseUrl>http.*\/courses\/\d+).*/);
-  return found.groups.courseUrl || null;
+  return found ? found.groups.courseUrl : url;
```
is this really the behavior we want? Is this a failure mode?

jxjj replied:
I was wondering too - what do you think? 
- Maybe better to default to the base url like?
- Or perhaps this should be handled in the chime component if null is returned, rather than in the selector? And then maybe just show a blank?

Leaning towards the latter as it's probably clearer... but not sure...

cmcfadden replied:
I lean towards the latter too. Since we can't think of a valid case where the previous line would fail and where we'd still want to vend the url, let's just fail to nothing.

jxjj replied:
works for me.

It'll probably fail if anyone's using ChimeIn with a different LMS... but they can file an issue and PR if those folks exist!

cmcfadden reviewed (approved):
good aside from my Q

jxjj commented:
Two changes to fix tests:
- removed the legacy factories packages. Laracasts cypress package tries to instantiate factories using the legacy way if it detects it.
- Changed default user permissions to 100 when created by factory. Forgot that permissions = 0 means no chime creation.
